JetMet validation for the Phase2

jdamgov opened the pull request:
JetMet validation (w/o DQM portion) for the Phase2 full reco.
This is only a temporary solution until all other sub-systems are prepared for running the standard validation

Dr15Jones committed:
Legacy and one modules must prefetch mayConsumes

In order to avoid the possibility of deadlocks occurring while a legacy or one module is doing a delayed get while running, we will disallow such activities. Once the full implementation of multiple threads per event is implemented, this restriction will be lifted.

Dr15Jones committed:
Always prefetch

To simplify the initial implementation of allowing multiple threads for one event, we require all 'mayConsumes' to be prefetched.

Dr15Jones committed:
Removed unnecessary functions

The Event based module signals are handled elsewhere.

Dr15Jones committed:
Make it clear which related test failed

Dr15Jones committed:
Add WaitingTaskList to InputProductResolver

Once we allow asynchronous data fetching from the source for one event we will need to using a WaitingTaskList to handle starting tasks which are waiting for the data.

Dr15Jones committed:
Refactoring needed to allow prefetching data asynchronously

-changed function names for prefetching to include Async and pass in a callback task as argument.
-separated prefetching and running modules into two different functions in Worker
-made sure all unit tests still pass

Dr15Jones committed:
Moved exceptionContext into Worker

exceptionContext was previously a standalone templated function. The template was being created for each OccurenceTraits. This was changed to just be a static non-templated function of Worker.

Dr15Jones committed:
Removed return value argument from ProductResolver::resolveProduct call

Previously one of the function arguments to ProductResolverBase::resolveProduct was used as a return value to denote if the request was 'ambiguous'. Such an argument is problematic for an asynchronous call. Instead, the return value of the function was changed to accommodate also carrying that information.

Dr15Jones committed:
Cache decisions from NoProcessProductResolver

Cache the decision of what ProductResolver to call after the first call each transition to NoProcessProductResolver. This will allow prefetching to trigger the real work and then when the data is later requested in the module we will not redo the work.

sethzenz committed:
code still messy

lgray committed:
sim tracks and sim clusters

lgray committed:
working to reconstruct caloparticle -> sim cluster hierarchy

lgray committed:
actually get the simclusters out

lgray committed:
caloparticles now store simcluster refs

Dr15Jones committed:
Begin transition to using SerialTaskQueues instead of mutexes

Ultimately shared resources will be handled via SerialTaskQueues but that transition can be done gradually. First start by using a SerialTaskQueue for all requests going to the Source.

Dr15Jones committed:
First crude implementation of asynchronous prefetching

This is only a starting point for asynchronous prefetches used to checkpoint that the code which has been written passes all framework unit tests.
Items which are still under work:
-Modules on Paths request asynchronous prefetches but then wait for them to be resolved.
-UnscheduledProductResolvers will run their modules synchronously if that is the first request for the data product. If other requests come while running the module, those requests will be done asynchronously.
-Mutexes are still used to synchronize across Streams.

lgray committed:
simclustering implemented

lgray committed:
SimCluster mapper (works for any detector) and PCA calculator for HGCal

lgray committed:
add simclusterizers for HGCal to local reconstruction path

Dr15Jones committed:
Make UnscheduledProductResolver::prefetchAsync actually asynchronous

Spawn a task in the case where prefetchAsync is called for the first time. This eventually should be changed to properly handle using the WaitingTaskQueue in the Worker as well as separating prefetching the products for the Worker from running of the module via the Worker.

Dr15Jones committed:
Added BusyWaitIntProducer for testing

The BusyWaitIntProducer was developed in order to have a test module actually take up some time and CPU cycles.

lgray committed:
trying to chase down some strange things in the event display (maybe geometry)

jhgoh committed:
update RPCRecHit and clustering dataformat to store time and y information

dgulhan committed:
rho analyzer

lgray committed:
fix a stupid bug in SimClusterMapper

Dr15Jones committed:
Updated unit test output for unscheduled changes

Updated the unit test output comparison files to account for the changed order of calling unscheduled modules. The order was changed because asynchronous prefetching runs the modules in FILO order instead of FIFO as was done before. The two orderings are equivalent since unscheduled doesn't care about the order.

dgulhan committed:
initialization added obsolete comments removed

lgray committed:
make parameters configurable

Dr15Jones committed:
Set worker state in a thread safe manner

Handle the worker state transitions as well as state accounting in a thread safe manner.
Switch to using std::exception_ptr for caching any exceptions.

dgulhan committed:
changed auto_ptr

Commit:
Changed HLT_IsoMu24_eta2p1 to HLT_IsoMu21_eta2p1

Conflicts:
	HLTriggerOffline/Btag/test/testSequences.py

Commit:
Changed HLT_IsoMu24_eta2p1 to HLT_IsoMu21_eta2p1

Commit:
Changed HLT_IsoMu24_eta2p1 to HLT_IsoMu21_eta2p1

Conflicts:
	HLTriggerOffline/Btag/python/HltBtagValidation_cff.py

Commit:
Changed HLT_IsoMu24_eta2p1 to HLT_IsoMu21_eta2p1

Conflicts:
	HLTriggerOffline/Btag/python/HltBtagPostValidation_cff.py

dgulhan committed:
changed to unique_ptr

Dr15Jones committed:
Added Worker::shouldRethrowException function

Moved the logic that decides if an exception should be rethrown into a separate function. This decreases the size of the templated Worker::doWork function and setups to allow that code to be used for the asynchronous processing function.

jhgoh committed:
assign nonzero y error

dgulhan committed:
rho producer

archiron committed:
PF isolation recomputation for miniAOD histos and minor others corrections

mmusich committed:
Updating Hcal MC pulse shapes

Commit:
Correct code for IsoTrack and PhiSymmetry calibration

mdhildreth committed:
Add capability to pass luminosity information to Digitizer, scale HIP rate

jhgoh committed:
do not ignore single-digi RPCRecHit to fill time information

Dr15Jones committed:
Separate prefetching and module running into different tasks

When running a module in unscheduled mode, have the UnscheduledProductResolver start a task to do any needed prefetching of data. Once all the prefetching has been completed, have the last task launch the task which will run the module.

Commit:
Add 2 new condition objects for HCAL: SiPMParameter and SiPMCharacteristics

archiron committed:
Float_t to float modification asked by davidlange6

lgray committed:
implement sim PF, working on handling electrons

Dr15Jones committed:
Added prefetching signals to ActivityRegistry

The Service system now emits a signal before and after a module has done prefetching.
The Tracer service was updated to use this signal.

Commit:
Fix a spelling

lgray committed:
address code review and get addons working again

archiron committed:
header simplification

lgray committed:
add SimPFProducer into HGCal era

Dr15Jones committed:
Enable Service system within TBB tasks

When a TBB task started by the framework starts up, we need to make sure the ServiceRegistry is properly setup to allow Services to be visible.

sdevissc committed:
first commit

Commit:
Move all files which uses SIMDataFormats to test

sdevissc committed:
couple of changes

sdevissc committed:
couple of changes

kpedro88 committed:
add run2_common to Phase2 Eras

kpedro88 committed:
improve localization of some phase2_hcal customs

kpedro88 committed:
eradicate tags hbheUpgradeReco and hfUpgradeReco from the face of the earth

Dr15Jones committed:
Make ROOT aware of all TBB threads

Now that we can use more TBB threads than Streams in the system, we need to make sure that all of the threads are known to ROOT. This is done by creating 1 task for each thread and then waiting until all tasks have done their work before proceeding.

lgray committed:
address further code review

sdevissc committed:
changing the APV flagging decision

lgray committed:
fix exception in CaloTruthAccumulator

kpedro88 committed:
reorganize upgrade workflow components

kpedro88 committed:
remove some unused upgrade steps

kpedro88 committed:
phase2 validation and dqm sequences

Dr15Jones committed:
Fixed case of recursive call into NoProcessProductResolver

Fixed the case where a module asks for data with the same label as
the module but from an earlier process via the skipCurrentProcess
mechanism.

lgray committed:
remove unneeded try/catch, guard with isValid()

kpedro88 committed:
new 2023 geometries and workflows

kpedro88 committed:
fix a few phase2 hcal things for global reco

kpedro88 committed:
temporarily demote hgcal workflows back to local reco

mdhildreth committed:
Merged AddingTheAffectedAPVList from repository sdevissc

mdhildreth committed:
Merge 810_pre9 version with vector to pass dead APVs

kpedro88 committed:
fix which upgrade workflows are exposed to regular matrix

wmtan committed:
Use unique_ptr, not auto_ptr

wmtan committed:
Use unique_ptr, not auto_ptr

Commit:
updating HBHEPulseShapeFlag and HBHEStatusBitSetter for Phase 1

jdamgov committed:
JetMet validation (no DQM) for the Phase2 upgrade full reco. A temporary solution

cmsbuild commented:
A new Pull Request was created by  @jdamgov  for CMSSW_8_1_X.

It involves the following packages:

Configuration/PyReleaseValidation
Validation/Configuration


@civanch, @vanbesien, @cvuosalo, @mdhildreth, @fabozzi, @cmsbuild, @srimanob, @slava77, @dmitrijus, @hengne, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @Martin-Grunewald this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

kpedro88 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 803e45232d47003a422ed8f1e3d8dc7b248e4b98
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15407/14452/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15407/14452/summary.html

There are some workflows for which there are errors in the baseline:
 50202.0 step 2
25202.0 step 2
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals.The error does NOT come from this pull request

davidlange6 committed:
Merge pull request #15357 from bsunanda/Run2-alca57

bsunanda:Run2-alca57 Add 2 new condition objects for HCAL: SiPMParameter and SiPMCharacteristics

mbandrews committed:
Fix LS reports: only use RawData.

makortel committed:
Remove DataFormats/TrackingSeed as obsolete

Dr15Jones committed:
Move CMS_THREAD_GUARD to proper line

CMS_THREAD_GUARD should be on the line for the variable being guarded and the argument should be the variable doing the guarding. The previous version of the code had those reversed.

mdhildreth committed:
Modify EventContent to save vector of affected APVs

kpedro88 committed:
fix typos

kpedro88 committed:
fix phase2 validation/dqm sequences

Dr15Jones committed:
Downgrade ROOT error from "Inverter::Dinv"

ROOT recently added an error message about matrix inversion failure.
Since the code returns a false if it fails, there is no need to convert
this to an exception. Instead we just forward to the MessageLogger.

Commit:
Bug fix. Carl, thanks for pointing this out.

mdhildreth committed:
Mods to make luminosity handling smoother

mdhildreth committed:
PreMixing modifications for HIP killing in DataMixer

cmsbuild committed:
Merge pull request #15297 from rockybala/Bugfix-for-HLT_IsoMu24_eta2p1

Bugfix for hlt iso mu24 eta2p1

cmsbuild committed:
Merge pull request #15335 from mmusich/81X_UpdateGTs_Jul31_2016

Update Hcal MC pulse shapes in all Global Tags

cmsbuild committed:
Merge pull request #15343 from archiron/New_Iso_Recomp_Histos_81X_V2

New PF iso recomputation from miniAOD histos 81X V2

davidlange6 committed:
Merge pull request #15404 from wmtan/UseUniquePtrInL1

Use unique_ptr, not auto_ptr, in 2 L1 packages

cmsbuild committed:
Merge pull request #15405 from wmtan/UseUniquePtrInDataFormatsMath

Use unique_ptr, not auto_ptr, in DataFormats/Math

cmsbuild committed:
Merge pull request #15414 from makortel/removeTrackingSeed

Remove DataFormats/TrackingSeed as obsolete

cmsbuild committed:
Merge pull request #15420 from Dr15Jones/downgradeRootErrorDinv

Downgrade ROOT error from "Inverter::Dinv"

mbandrews committed:
Re-enable presample in global qt

cmsbuild committed:
Merge pull request #15293 from lgray/topic_simclustering

SimClustering for HGCal

cmsbuild committed:
Merge pull request #15354 from bsunanda/Run2-alca56

bsunanda:Run2-alca56 Correct code for IsoTrack and PhiSymmetry calibration

cmsbuild committed:
Merge pull request #15263 from jhgoh/RPC2D

RPCRecHit with timing and y-coordinate information for iRPC

dgulhan committed:
clone kt4 jets

Commit:
MC updates for tranceIV - see: https://indico.cern.ch/event/562065/contributions/2270369/

Commit:
MC updates for tranceIV - see: https://indico.cern.ch/event/562065/contributions/2270369/ (v5)

cmsbuild committed:
Merge pull request #15406 from igv4321/filter-update

Updating HBHEPulseShapeFlag and HBHEStatusBitSetter for Phase 1

Dr15Jones committed:
Added missing header files needed for gcc 6

The gcc 6 compilation was failing without explicit inclusion of additional C++ headers.

davidlange6 committed:
Merge pull request #15435 from Dr15Jones/fixGcc6Compilation

Added missing header files needed for gcc 6

davidlange6 committed:
Merge pull request #15433 from Dr15Jones/prefetchConcurrently

Prefetch data products from Event concurrently

davidlange6 committed:
Merge pull request #15393 from kpedro88/Phase2-WF12

Phase2 workflow revamp

kpedro88 commented:
@jdamgov #15393 is now merged; once it is in an IB (check https://cmssdt.cern.ch/SDT/html/showIB.html), you can rebase on top of it. The modification to `relval_steps.py` is no longer necessary; just add the name of the sequence [here](https://github.com/cms-sw/cmssw/blob/72363198bc82295bdf87793fde7298ef5916dd5f/Validation/Configuration/python/autoValidation.py#L9).

mdhildreth committed:
Final cleanup

mdhildreth committed:
Final final cleanup

Dr15Jones committed:
Minor maintenance improvements to WaitingTaskList

-Used C++11 more effectively
-added CMS_THREAD_GUARD for annotating intent
-created make_empty_waiting_task to avoid copy and paste errors and to switch to std::unique_ptr instead of std::shared_ptr to avoid an unnecessary call to new.

davidlange6 committed:
Merge pull request #15438 from Dr15Jones/cleanupWaitingTaskList

Minor maintenance improvements to WaitingTaskList

davidlange6 committed:
Merge pull request #15432 from cms-AlCaDB/gf-AlCaForTrancheIV-from-CMSSW_8_1_X

MC updates for trancheIV

cmsbuild committed:
Merge pull request #15233 from dgulhan/RhoForpPbRelv3

rho analyzer

davidlange6 committed:
fix bug that led to double inclusion of customise from command line

davidlange6 committed:
Merge pull request #15443 from davidlange6/CBDouble2

fix bug that led to double inclusion of customise from command line in cmsDriver

cmsbuild committed:
Merge pull request #15412 from mbandrews/LSReport81X

Fix LS reports: only use RawData.

mmusich committed:
Adding vtx smearing parameters for 2015 Beta*=90m for FSQ analyes

Dr15Jones committed:
Use std::move in SharedResourcesAcquirer constructor

Although the first argument to the constructor is an lvalue reference, it is still required to use std::move to get the benefits.

mdhildreth committed:
fix remaining issues

mdhildreth committed:
stupid override

Dr15Jones committed:
Added additional comments

Dr15Jones committed:
Properly handle non CMS exceptions in Worker::runModuleAfterAsyncPrefetch

Moved code into the convertException::wrap just in case a non-CMS exception was thrown.

Dr15Jones committed:
Use sentry to guarantee ActivityRegistry signal happens

Use an std::unique_ptr to guarantee that at then end of the code block the postModuleEventPrefetchSignal will be emitted, even if an exception happens.

davidlange6 committed:
Merge pull request #15456 from Dr15Jones/useMoveInSharedResourcesAcquirer

Small modifications based on code review of asynchronous prefetching

davidlange6 committed:
Merge pull request #15452 from mmusich/81X_betaStar90m

Adding vtx smearing parameters for 2015 Beta*=90m for FSQ analyses

davidlange6 committed:
Merge pull request #15436 from mdhildreth/SiStripAPVVec

HIP Killing and PreMixing for 81X

mmusich committed:
updating SiStrip DQM TriggerBits for HIP monitoring (PR #15447)

cmsbuild committed:
Merge pull request #15460 from mmusich/81X_updateGTs_forHIPMonitoring

Updating SiStrip DQM TriggerBits for HIP monitoring (PR #15447)

jdamgov committed:
rebase to IB CMSSW_8_1_X_2016-08-14-2300.

Merged Phase2_JetMet_Validation_v1 from repository jdamgov

Conflicts:
	Configuration/PyReleaseValidation/python/relval_steps.py

cmsbuild commented:
Pull request #15407 was updated. @ghellwig, @ianna, @cerminar, @fabozzi, @Martin-Grunewald, @rekovic, @vanbesien, @perrotta, @civanch, @franzoni, @cmsbuild, @fwyzard, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @mdhildreth, @slava77, @ggovi, @mmusich, @mulhearn, @dmitrijus, @alja, @srimanob, @hengne can you please check and sign again.

jdamgov commented:
Superseded by PR15466
